src/clr-angular/data/datagrid/providers/column-toggle-buttons.service.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {Injectable, TemplateRef} from "@angular/core";
+import {Observable} from "rxjs/Observable";
+import {Subject} from "rxjs/Subject";
+
+export type ColumnToggleButtons = "ok"|"selectAll";
+
+@Injectable()
+export class ColumnToggleButtonsService {
+    buttons: TemplateRef<any> = null;
+    selectAllDisabled: boolean = false;
+
+    private _okButtonClicked = new Subject<any>();
+    public get okButtonClicked(): Observable<any> {
+        return this._okButtonClicked.asObservable();
+    }
+
+    private _selectAllButtonClicked = new Subject<any>();
+    public get selectAllButtonClicked(): Observable<any> {
+        return this._selectAllButtonClicked.asObservable();
+    }
+
+    public buttonClicked(type: ColumnToggleButtons): void {
+        switch (type) {
+            case "ok":
+                this._okButtonClicked.next();
+                break;
+            case "selectAll":
+                this._selectAllButtonClicked.next();
+                break;
+            default:
```

Comment:
I suspect the answer is "Yes, because JavaScript" but if we have a custom type that requires two strings (either "ok" or "selectAll") And we create a case statement that covers both of those types, do we still need the default fall through?

Reply:
I believe we need it because tslint requires us to have it.

Reply:
Yeah, tooling doesn't like no `default` case, and I _could_ have it use one of the two as default but then it makes sense to just use if/else.